lib/configuration.ts

```diff
@@ -0,0 +1,56 @@
+import { getEnvVars, setEnvVars } from "./environment";
+import { createClient } from "./graphql";
+import prisma from "./prisma";
+
+export interface Configuration {
+  domain: string;
+  token: string;
+}
+
+const MT_MODE = true;
+
+export const getConfiguration = async (domain: string) => {
+  if (MT_MODE) {
+    const registration = await prisma.appRegistration.findFirst({
+      where: { domain },
+    });
+    if (!registration) {
+      throw new Error(`Domain ${domain} not registered`);
+    }
+    return registration;
+  } else {
+    const env = await getEnvVars();
+    if (domain !== env.SALEOR_DOMAIN) {
+      throw new Error(`Domain ${domain} not registered`);
+    }
+    return {
+      token: env.SALEOR_AUTH_TOKEN,
+      domain: env.SALEOR_DOMAIN,
+    };
+  }
+};
+
+export const setConfiguration = async (configuration: Configuration) => {
+  if (MT_MODE) {
+    await prisma.appRegistration.create({
+      data: configuration,
+    });
+  } else {
+    await setEnvVars([
+      {
+        key: "SALEOR_AUTH_TOKEN",
+        value: configuration.token,
+      },
+      {
+        key: "SALEOR_DOMAIN",
+        value: configuration.domain,
+      },
+    ]);
+  }
+};
+
+export const saleorApiClient = (saleorDomain: string) => {
+  return createClient(`https://${saleorDomain}/graphql/`, async () =>
+    Promise.resolve({ token: (await getConfiguration(saleorDomain))?.token })
```

Comment:
How long token lives? Can we/should we cache it for speed? 
Ofc it's not critical for MVP

Reply:
App tokens don't expire, so once generated we can use them for all eternity 😉

Reply:
I'd be careful with that, probably we should be able to invalidate token due to security reasons 

@krzysztofwolski ideas?